### PR TITLE
Added a first vtk output for tensor-wedges

### PR DESCRIPTION
**Pull request by Davknapp**
I develop a vtk-output for TensorProductWedges. 
This type of vtk-output is not based on Lagrange-elements, because the nodes of the TensorProductWedge can be distributed very different from the the nodes of a VTK_LAGRANGE_WEDGE.
The nodes of the triangulare basis define a subdivision of the wedge, so does the line. Given these subdivisions, wedges are defined and written into the vtk-file.

### CODE REVIEW

File: src/StartUpDG.jl [hunk 1]
```diff
@@ -79,7 +79,7 @@ export read_HOHQMesh
 # Plots.jl recipes for meshes
 include("mesh/vtk_helper.jl")
 include("mesh/mesh_visualization.jl")
-export VertexMeshPlotter, MeshPlotter, MeshData_to_vtk
+export VertexMeshPlotter, MeshPlotter, MeshData_to_vtk, MeshData_Tensor_to_vtk
```
Comment (jlchan) [inline src/StartUpDG.jl:82]: See comment below on dispatch

File: src/mesh/mesh_visualization.jl [hunk 2]
```diff
@@ -113,3 +113,111 @@ function MeshData_to_vtk(md::MeshData, rd::RefElemData{DIM}, data, dataname, fil
...
+
+Translate the given mesh into a vtk-file.
+`md` holds a `MeshData` object
+`rd` holds a reference element data/`RefElemData` of a TensorProductWedge
+`data` holds an array of arrays (of size `num_nodes` by `num_elements`) with plotting data
+`dataname` is an array of strings with name of the associated data
+`write_data`, flag if data should be written or not (e.g., if data is not written, only the mesh will be saved as output)
+`equi_dist_nodes` flag if points should be interpolated to equidstant nodes
+"""
+function MeshData_Tensor_to_vtk(md::MeshData, rd::RefElemData{DIM}, data, dataname, filename,
```
[the remark this answers is not in the document]
Comment (Davknapp) [reply src/mesh/mesh_visualization.jl]: Nevermind, figured it out! We had the same difficulties in a Trixi-PR, where you found the solution for that problem. Thanks ;)

File: src/mesh/mesh_visualization.jl [hunk 3]
```diff
@@ -113,3 +113,111 @@ function MeshData_to_vtk(md::MeshData, rd::RefElemData{DIM}, data, dataname, fil
...
+
+Translate the given mesh into a vtk-file.
+`md` holds a `MeshData` object
+`rd` holds a reference element data/`RefElemData` of a TensorProductWedge
+`data` holds an array of arrays (of size `num_nodes` by `num_elements`) with plotting data
+`dataname` is an array of strings with name of the associated data
+`write_data`, flag if data should be written or not (e.g., if data is not written, only the mesh will be saved as output)
+`equi_dist_nodes` flag if points should be interpolated to equidstant nodes
+"""
+function MeshData_to_vtk(md::MeshData, rd::RefElemData{3, <:Wedge, <:TensorProductWedge}, data, dataname, filename,
```
Comment (jlchan) [inline src/mesh/mesh_visualization.jl:127]: Oops, my suggestion forgot one parameter. Thanks for catching and fixing this!

File: src/mesh/mesh_visualization.jl [hunk 4]
```diff
@@ -113,3 +113,111 @@ function MeshData_to_vtk(md::MeshData, rd::RefElemData{DIM}, data, dataname, fil
     return vtk_save(vtkfile)
 end
 
+"""
+MeshData_Tensor_to_vtk(md, rd, dim, data, dataname, datatype, filename, write_data = false, equi_dist_nodes = true)
```
Comment (jlchan) [inline src/mesh/mesh_visualization.jl:117]: Can you update this docstring?

File: src/mesh/mesh_visualization.jl [hunk 5]
```diff
@@ -113,3 +113,111 @@ function MeshData_to_vtk(md::MeshData, rd::RefElemData{DIM}, data, dataname, fil
...
+            for i in 1:line_inter_size
+                z_equi = vcat(z_equi, fill(z_tmp[i], tri_inter_size))
+            end
+            coords[3][:, elem] = z_equi
+        end
+        coords = vec.(coords)  
+    else
+        coords = vec.(md.xyz)
+    end
+    vtkfile = vtk_grid(filename, coords...,cells)
```
Comment (jlchan) [inline src/mesh/mesh_visualization.jl:213]:
```suggestion
    vtkfile = vtk_grid(filename, coords..., cells)
```

File: src/mesh/mesh_visualization.jl [hunk 6]
```diff
@@ -113,3 +113,111 @@ function MeshData_to_vtk(md::MeshData, rd::RefElemData{DIM}, data, dataname, fil
...
+        # equi-distant nodes for triangular basis (hence only concider x-y-coords. )
+        for dim in 1:2
+            # iterate over all elements
+            for elem in 1:md.num_elements
+                tmp_coords = Vector{Float64}[]
+                for i in 1:line_inter_size
+                    # interpolate each slice of the wedge
+                    tmp_coords = vcat(tmp_coords, tri_interpolate * md.xyz[dim][((i-1)*tri_inter_size + 1):(i*tri_inter_size), elem])
+                end
+                coords[dim][:, elem] =  tmp_coords
```
[the remark this answers is not in the document]
Comment (Davknapp) [reply src/mesh/mesh_visualization.jl]: I think I already found a solution for it, see the last pushes.

**Comment by Davknapp** [comment]
> LGTM! Any other changes you want to make before we merge?

Thanks! No, nothing to add currently.

**Comment by jlchan** [comment]
Thanks again for adding this! I'll release a new version

**Pull request merged**